common/src/View/CameraTool3D.cpp

```diff
@@ -154,6 +154,41 @@ void CameraTool3D::mouseScroll(const InputState& inputState)
 {
   const float factor = pref(Preferences::CameraMouseWheelInvert) ? -1.0f : 1.0f;
   const bool zoom = inputState.modifierKeysPressed(ModifierKeys::MKShift);
+
+  if (pref(Preferences::CameraTrackpadMode))
+  {
+    const auto orbit =
+      inputState.modifierKeysPressed(ModifierKeys::MKAlt) || inputState.scrollW() != 0.0f;
+
+    const auto moveDirection = pref(Preferences::CameraMoveInCursorDir)
+                                 ? vm::vec3f{inputState.pickRay().direction}
+                                 : m_camera.direction();
+
+    if (orbit)
+    {
+      m_camera.orbit(
+        vm::vec3f{m_camera.defaultPoint(inputState.pickRay())},
```

Comment:
Originally I was applying the same hit testing used for the drag tracker here, only without storing the orbit point across the whole interaction. It was producing weird results (since another brush could appear "in front" of the orbit point and capture the orbit) so I removed it.

We can store the orbit point on the `CameraTool3D` instance, but I'm worried whether or not that is clean architecturally. Let me know if you'd like me to do that, or something else.

Reply:
Yeah, this is a bit unfortunate. I introduced `DragTracker` to avoid storing any data associated with gestures in the tools themselves. I suggested introducing a notion of gestures in the previous commit. I think we might then also want to generalize `DragTracker` to `GestureTracker` and generalize its interface to have `start/update/stop` semantics. Then we could use that for trackpad gestures as well as dragging gestures.